contributors/design-proposals/device-plugin.md

```diff
@@ -0,0 +1,471 @@
+# Device Manager Proposal
+
+  1. [Abstract](#abstract)
+  2. [Motivation](#motivation)
+  3. [Use Cases](#use-cases)
+  4. [Objectives](#objectives)
+  5. [Non Objectives](#non-objectives)
+  6. [Stories](#stories)
+      * [Vendor story](#vendor-story)
+      * [User story](#user-story)
+  8. [Device Plugin](#device-plugin)
+      * [Protocol Overview](#protocol-overview)
+      * [Protobuf specification](#protobuf-specification)
+      * [Installation](#installation)
+      * [API Changes](#api-changes)
+      * [Versioning](#versioning)
+
+_Authors:_
+
+* @RenaudWasTaken - Renaud Gaubert &lt;rgaubert@NVIDIA.com&gt;
+
+## Abstract
+
+This document describes a vendor independant solution to:
+  * Discovering and representing external devices
+  * Making these devices available to the container and cleaning them up
+    afterwards
+  * Health Check of these devices
```

Comment:
noting here, but i would like to also understand how i perform basic node life-cycle operations:

1. drain of a node that was using these plugins (does ordering matter)
2. upgrade of host, say a security vulnerability (i.e. yum update, etc.)
3. upgrade to new kubectl versions (what concerns, if any do i have with skew)